xdsl/parser/core.py

```diff
@@ -792,6 +797,32 @@ def _parse_op_result_list(self) -> list[tuple[Span, int]]:
     def parse_optional_attr_dict(self) -> dict[str, Attribute]:
         return self.parse_optional_dictionary_attr_dict()
 
+    def resolve_operands(
+        self, args: list[UnresolvedOperand], input_types: ArrayAttr[Attribute]
+    ) -> list[SSAValue]:
+        """
+        Resolve unresolved operands. For each operand in `args` and its corresponding input
+        type the following happens:
+
+        If the operand is not yet defined, it creates a forward reference.
+        If the operand is already defined, it returns the corresponding SSA value,
+        and checks that the type is consistent.
+
+        If the length of args and input_types does not match, an error is raised at
+        the current location.
+        """
+        func_type_pos = self._current_token.span.start
```

Comment:
This feels like a surprising coupling of the state of the parser with the behaviour of this function, which only declares inputs and outputs in the type signature. When called by clients outside of the parser, is it really expected that they call it when in a state where the current token is relevant to the error message? I have a feeling that clients might call this method once they've parsed the entire line, so the token wouldn't really make sense.

Reply:
Indeed. I got tempted but this is clearly not the right approach.